src/prefect/events/schemas/automations.py

```diff
@@ -429,4 +432,110 @@ class AutomationCore(PrefectBaseModel, extra="ignore"):
 
 
 class Automation(AutomationCore):
-    id: UUID = Field(..., description="The ID of this automation")
+    id: Optional[UUID] = Field(default=None, description="The ID of this automation")
+
+    @classmethod
+    @sync_compatible
+    async def create(cls: Self, automation: Type[Self]) -> UUID:
+        """
+        Create a new automation.
+
+        auto_to_create = Automation(
+        name="woodchonk",
+        trigger=EventTrigger(
+            expect={"animal.walked"},
+            match={
+                "genus": "Marmota",
+                "species": "monax",
+            },
+            posture="Reactive",
+            threshold=3,
+            within=timedelta(seconds=10),
+        ),
+        actions=[CancelFlowRun()]
+        )
+        created_automation = Automation.create(auto_to_create)
+        """
+        client, _ = get_or_create_client()
+        automation = AutomationCore(**automation.dict(exclude={"id"}))
+        automation = await client.create_automation(automation=automation)
+        return automation if automation else None
+
+    @classmethod
+    @sync_compatible
+    async def update(cls: Self, automation_id: UUID, automation: Type[Self]) -> UUID:
```

Comment:
It feels a little weird to me that this is a class method and that you can pass a separate ID and automation like this.  I'd expect it more like:

```
automation = Automation.create(....)
automation.trigger.posture = Posture.Proactive
automation.update()  # <-- instance method with no arguments
```

Like, what would happen if you passed the ID of a different automation (it would get overridden) or a new ID (you'd get a 404 error).  I'd make this an instance method and use `self.id` for the `client.update_automation(...)` call.  Same for `delete`, `enable`, and `disable`, I think.

Reply:
cool just to be clear here - automation.create doesnt current return an automation object, just the UUID because thats what the API returns. I am going to change it to make an addition read call to grab up that object at the end and then all this will work as written above and lmk if feedback on that once its up.